link/executable.go

```diff
@@ -0,0 +1,78 @@
+package link
+
+import (
+	"debug/elf"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/cilium/ebpf/internal"
+)
+
+var ErrSymbolNotFound error
```

Comment:
Does this have to be exported? If yes, it needs documentation and the function that may return it should also document that fact.

Reply:
can be a generic `error`, will change it